src/kernel/src/machine/arm/morello/info.rs

```diff
@@ -0,0 +1,123 @@
+use fdt::Fdt;
+use twizzler_abi::device::{CacheType, MmioInfo};
+
+use crate::{arch::BootInfoSystemTable, once::Once, BootInfo};
+
+// We use device tree to describe the hardware on this machine
+static FDT: Once<Fdt<'static>> = Once::new();
+
+pub fn init<B: BootInfo>(boot_info: &B) {
+    FDT.call_once(|| {
+        let dtb = {
+            // try to find device tree location using the bootloader
+            let bootloader_dtb_addr = boot_info.get_system_table(BootInfoSystemTable::Dtb);
+            // otherwise use a static address
+            if bootloader_dtb_addr.raw() == 0 {
+                // in the case of QEMU's virt platform, we can use 0x4000_0000
+                super::memory::DTB_ADDR.kernel_vaddr()
+            } else {
+                bootloader_dtb_addr
+            }
+        };
+        // should not fail, but it might ...
```

Comment:
Can you add a bit more commentary here?  Why might it?  This is a very enigmatic comment.